mycli/main.py

```diff
@@ -1007,6 +1011,22 @@ def need_completion_refresh(queries):
         except Exception:
             return False
 
+
+def is_dropping_database(queries, dbname):
+    """Determine if the query is dropping a specific database."""
+    for query in sqlparse.split(queries):
+        try:
+            stmt = sqlparse.parse(query)[0]
+            first_token = stmt.token_first(skip_cm=True)
+            _, second_token = stmt.token_next(0, skip_cm=True)
+            if (first_token.value.lower() == 'drop' and
+                    second_token.value.lower() in ('database', 'schema') and
+                    stmt.get_name().lower() == dbname.lower()):
+                return True
+        except Exception:
```

Comment:
I don't know if we need to disable the catch all Exceptions for debugging. What if we just try to figure out what the appropriate exception is and catch that instead of catching all exceptions? I've noticed a few of those around mycli.